File: C&Python/Stack&Queue/프린터.py
from queue import Queue


def solution(priorities, location):
    # 초기화
    answer = 1
    k = 0
    que = Queue()
    sort_list = []
    find_value = priorities[location]
    find_location = location
    
    # queue, 정렬 list insert(튜플 형태)
    for i in range(len(priorities)):
        que.put((priorities[i], i))
        sort_list.append(priorities[i])
    
    # 정렬 list sort
    sort_list.sort(reverse=True)
    
    while que.empty() == False:     
        # queue의 front 값 pop
        max_value = sort_list[k]
        get_value, get_index = que.get()
        
        # 우선순위 가장 큰가? and 요청한 값인가? => true면 return
        if get_value == max_value:
            if get_value == find_value and get_index == find_location:
                return answer
            answer += 1
            k += 1
        else:
            # 다시 queue에 push
            que.put((get_value, get_index))
    
    # 찾는 값이 없는 경우.
    return -1 

File: C++/Programmers/Stack,Queue/프린터.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <queue>
using namespace std;
bool cmp(int a, int b)
{
	return a > b;
}
int solution(vector<int> priorities, int location) {
	int answer = 0;
	vector <int> order = priorities;
	queue<pair<int, int>> q;
	for (int i = 0; i < priorities.size(); i++)
	{
		pair <int, int> temp;
		temp = make_pair(priorities[i], i);
		q.push(temp);
	}
	sort(order.begin(), order.end(), cmp);
	vector<int> ::iterator iter;
	iter = order.begin();
	while (!q.empty())
	{
		if (q.front().first < *iter)
		{
			q.push(q.front());
			q.pop();
		}
		else
		{
			if (q.front().second == location)
			{
				answer++;
				break;
			}
			else
			{
				q.pop();
				answer++;
			}
			iter++;
		}
	}
	return answer;
}